### PR TITLE
Reporting threads and comments

### PR DESCRIPTION
**Pull request by hdemusg**
For threads, the report links to the thread to delete/unreport.
For comments, the report has buttons to handle the deletion/reporting immediately.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,6 +58,7 @@
     "node-sass": "^4.14.1",
     "prop-types": "^15.7.2",
     "react": "^16.13.1",
+    "react-bootstrap": "^1.4.0",
```
Comment (KanavKG) [inline package.json:61]: Why did you include bootstrap?

File: server/mongodb/actions/AskMeThread.js [hunk 2]
```diff
@@ -27,6 +27,72 @@ export const createThread = async (
...
+    .then(async (reportedThread) => {
+      if (reportedThread == null) {
+        throw new Error(
+          "No thread matches the provided id or user does not have permission!"
+        );
+      }
+    });
+};
+
+// export const unreportAskMeThread = async (currentUser, { id }) => {
```
Comment (KanavKG) [inline server/mongodb/actions/AskMeThread.js:73]: Remove all comments please.

File: server/mongodb/actions/Comment.js [hunk 3]
```diff
@@ -100,3 +122,36 @@ export const getCommentsByThread = async (currentUser, { id }) => {
...
+};
+
+export const unreportComment = async (currentUser, { id }) => {
+  if (currentUser == null || id == null) {
+    throw new Error("All parameters must be provided!");
+  }
+
+  await mongoDB();
+
+  //const query = { _id: id };
```
Comment (KanavKG) [inline server/mongodb/actions/Comment.js:145]: Remove comments

File: server/mongodb/actions/GroupThread.js [hunk 4]
```diff
@@ -19,14 +19,57 @@ export const createThread = async (
...
+};
+
+export const unreportGroupThread = async (currentUser, { id }) => {
+  if (currentUser == null || id == null) {
+    throw new Error("All parameters must be provided!");
+  }
+
+  await mongoDB();
+
+  //const query = { _id: id };
```
Comment (KanavKG) [inline server/mongodb/actions/GroupThread.js:59]: Remove comments

File: server/mongodb/models/AskMeThread.js [hunk 5]
```diff
@@ -44,11 +54,42 @@ async function handleDelete(provDoc) {
...
 
+async function handleReport(provDoc) {
+  const doc =
+    this.getQuery != null ? await this.find(this.getQuery()) : provDoc;
+
+  if (doc != null) {
+    const id = doc._id;
+    if (!this.reported) {
+      await this.update({ _id: id }, { reported: true });
+      // await mongoose
```
Comment (KanavKG) [inline server/mongodb/models/AskMeThread.js:65]: Remove comments

File: server/mongodb/models/AskMeThread.js [hunk 6]
```diff
@@ -44,11 +54,42 @@ async function handleDelete(provDoc) {
...
+
+async function handleUnreport(provDoc) {
+  const doc =
+    this.getQuery != null ? await this.find(this.getQuery()) : provDoc;
+
+  if (doc != null) {
+    const id = doc._id;
+    if (this.reported) {
+      await this.update({ _id: id }, { reported: false });
+      // await mongoose
```
Comment (KanavKG) [inline server/mongodb/models/AskMeThread.js:79]: Remove comments

File: server/mongodb/models/GroupThread.js [hunk 7]
```diff
@@ -39,7 +49,22 @@ async function handleDelete(provDoc) {
...
+
+async function handleReport(provDoc) {
+  const doc =
+    this.getQuery != null ? await this.find(this.getQuery()) : provDoc;
+
+  if (doc != null) {
+    const id = doc._id;
+    if (!this.reported) {
+      await this.update({ _id: id }, { reported: true });
+      // await mongoose
```
Comment (KanavKG) [inline server/mongodb/models/GroupThread.js:64]: Remove comments

File: src/components/Admin/Reports/AskMeReport/index.jsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    <a className={styles.AskMeReport}>
+      <div className={styles.ReportHeader}>
+        <h3 className={styles.Question}>{`Question: ${Thread.title}`}</h3>
+        <button variant="success">Go to Thread to Ignore or Delete</button>
+      </div>
+    </a>
+  </Link>
+);
+
+//return bookmarks.map((question) => (
```
Comment (KanavKG) [inline src/components/Admin/Reports/AskMeReport/index.jsx:23]: Remove comments

File: src/components/Admin/Reports/CommentReport/index.jsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      <div className={styles.ReportHeader}>
+        <h3 className={styles.Question}>{`Comment: ${Comment.content}`}</h3>
+        <button variant="success" onClick={unreport}>Approve</button>
+        <button variant="danger" onClick={del}>Delete</button>{" "}
+      </div>
+    </a>
+   );  
+ };
+
+//return bookmarks.map((question) => (
```
Comment (KanavKG) [inline src/components/Admin/Reports/CommentReport/index.jsx:34]: Remove comments

File: src/components/Admin/Reports/GroupReport/index.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      <div className={styles.ReportHeader}>
+        <h3 className={styles.Question}>{`Question: ${Thread.title}`}</h3>
+        <button variant="success">Go to Thread to Ignore or Delete</button>
+      </div>
+    </a>
+  </Link>
+);
+
+//return bookmarks.map((question) => (
+// <QuestionCard key={question._id} question={question} />
```
Comment (KanavKG) [inline src/components/Admin/Reports/GroupReport/index.jsx:24]: Remove comments

**Review by KanavKG — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (hdemusg) [reply package.json:61]: We were trying to add styling for buttons to make them green/red. This is not successful yet.

**Pull request merged**